cluster-provision/okd/scripts/provision.sh

```diff
@@ -208,13 +208,16 @@ oc adm policy add-cluster-role-to-user cluster-admin admin
 
 if [ "${CNAO}" == "true" ]; then
     # Apply network addons
-    oc create -f /manifests/cna/namespace.yaml
-    oc create -f /manifests/cna/network-addons-config.crd.yaml
-    oc create -f /manifests/cna/operator.yaml
-    oc create -f /manifests/cna/network-addons-config-example.cr.yaml
 
-     # Wait until all the network components are ready
-    until oc wait networkaddonsconfig cluster --for condition=Available --timeout=100s; do
+    # TODO: pass CNAO version at gocli
+    cnao_version=0.25.0
+    cnao_path=/manifests/cnao/$cnao_version/
+    oc create -f $cnao_path/namespace.yaml
+    oc create -f $cnao_path/network-addons-config.crd.yaml
+    oc create -f $cnao_path/operator.yaml
+
+     # Wait until cnao is ready
+    until oc wait deployment -n cluster-network-addons cluster-network-addons-operator --for condition=Available --timeout=100s; do
         sleep 10
```

Comment:
The timeout is not long enough, I would set it to 5 minutes, lot of images are pulled in the process.

Also, there is no need for `until`, just use:

```
oc wait deployment -n cluster-network-addons cluster-network-addons-operator --for condition=Available --timeout=5m
```